tests/hazmat/primitives/test_dsa.py

```diff
@@ -40,6 +41,13 @@ def _skip_if_dsa_not_supported(backend, algorithm, p, q, g):
         )
 
 
+def _min_key_size(backend):
+    if backend._fips_enabled:
+        return backend._dh_min_key_size
+    else:
+        return 1024
```

Comment:
Is there a reason not to always just use 2048 here?

Reply:
Generating a 2048-bit DH key is annoyingly slow, but avoiding the branch is probably worth it.